.travis.yml

```diff
@@ -13,6 +13,10 @@ env:
   global:
     - PYTEST_ADDOPTS=-vv
 
+# setuptools-scm needs all tags in order to obtain a proper version
+git:
+  depth: false
```

Comment:
You could still limit it to 1000.
The default is 50.
Having additional merge commits for single commit changes (or even 5+ "updated X" commits for what should be a single commit, because you cannot easily squash-merge certainly does not help here ;))

But to be fair: for a stable branch it is easy to get away 50+ commits anyway.